x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -11,16 +11,29 @@ import (
 // TODO add unit tests for all the keepers
 
 // GetCurrentDataCommitment creates the latest data commitment at current height according to
-// the data commitment window specified
+// the data commitment window specified.
+// Assumes that this method is called when:
+//
+//	ctx.BlockHeight() != 0 && ctx.BlockHeight()%int64(k.GetDataCommitmentWindowParam(ctx)) == 0
 func (k Keeper) GetCurrentDataCommitment(ctx sdk.Context) (types.DataCommitment, error) {
-	beginBlock := uint64(ctx.BlockHeight()) - k.GetDataCommitmentWindowParam(ctx) + 1
-	// for a data commitment window of 400, the ranges will be: 1-400;401-800;801-1200
-	endBlock := uint64(ctx.BlockHeight())
-
 	if !k.CheckLatestAttestationNonce(ctx) {
 		return types.DataCommitment{}, types.ErrLatestAttestationNonceStillNotInitialized
 	}
 	nonce := k.GetLatestAttestationNonce(ctx) + 1
+	dcWindow := k.GetDataCommitmentWindowParam(ctx)
+	// for a data commitment window of 400, the ranges will be: 1-400;401-800;801-1200
+	var beginBlock, endBlock uint64
+	if ctx.BlockHeight() == int64(dcWindow) {
+		beginBlock = 1
+		endBlock = dcWindow
```

Comment:
maybe add a doc just to clarify that this is only for the first data commitment

Reply:
[36c4523](https://github.com/celestiaorg/celestia-app/pull/1695/commits/36c45239776af731f97e22bf0b35ec83a1ea2c98)